sky/core.py

```diff
@@ -888,6 +891,80 @@ def spot_tail_logs(name: Optional[str], job_id: Optional[int],
     backend.tail_spot_logs(handle, job_id=job_id, job_name=name, follow=follow)
 
 
+@usage_lib.entrypoint
+def spot_cost_report(refresh: bool, verbose: bool) -> List[Dict[str, Any]]:
+
+    # NOTE(dev): Keep the docstring consistent between the Python API and CLI.
+    """Get statuses of managed spot jobs.
+
+    Please refer to the sky.cli.spot_queue for the documentation.
+
+    Returns:
+        [
+            {
+                'job_id': int,
+                'job_name': str,
+                'resources': str,
+                'submitted_at': (float) timestamp of submission,
+                'end_at': (float) timestamp of end,
+                'duration': (float) duration in seconds,
+                'retry_count': int Number of retries,
+                'status': sky.JobStatus status of the job,
+                'cluster_resources': (str) resources of the cluster,
+                'region': (str) region of the cluster,
+            }
+        ]
+    Raises:
+        sky.exceptions.ClusterNotUpError: the spot controller is not up.
+    """
+
+    stop_msg = ''
+    if not refresh:
+        stop_msg = 'To view the latest job table: sky spot cost --refresh'
+
+    controller_status, handle = spot.is_spot_controller_up(stop_msg)
+
+    if controller_status is None:
+        return []
+
+    if (refresh and controller_status in [
+            global_user_state.ClusterStatus.STOPPED,
+            global_user_state.ClusterStatus.INIT
+    ]):
+        print(f'{colorama.Fore.YELLOW}'
+              'Restarting controller for latest status...'
+              f'{colorama.Style.RESET_ALL}')
+
+        handle = _start(spot.SPOT_CONTROLLER_NAME)
+
+    controller_status = global_user_state.ClusterStatus.UP
+
+    if handle is None or handle.head_ip is None:
+        raise exceptions.ClusterNotUpError('Spot controller is not up.',
+                                           cluster_status=controller_status)
+
+    backend = backend_utils.get_backend_from_handle(handle)
+    assert isinstance(backend, backends.CloudVmRayBackend)
+
+    code = spot.SpotCodeGen.get_cost_report(verbose)
```

Comment:
This is open for discussion: should we do the aggregation on the controller or should we only provide `get_cost_report` on the controller and do the aggregation locally, if needed?